### PR TITLE
harden ES service detection to provide nicer UX

### PR DESCRIPTION
**Pull request by pkarman**
trello: https://trello.com/c/uFwAucpL/185-search-service-responds-with-502

Creates a custom exception class for when ES service goes AWOL.

### CODE REVIEW

File: app/controllers/proposals_controller.rb [hunk 1]
```diff
@@ -61,7 +61,12 @@ def complete
   def query
     check_search_params
     query_listing = listing
-    @proposals_data = query_listing.query
+    begin
```
Comment (yozlet) [inline app/controllers/proposals_controller.rb]: Should we have this in `#download` as well (and possibly the report generation) or is that overkill?

**Comment by yozlet** [comment]
Other than the question above, :+1:

File: app/controllers/proposals_controller.rb (hunk 1 above)
Comment (pkarman) [inline app/controllers/proposals_controller.rb]: good idea, @yozlet -- added in 2219a96

**Comment by jessieay** [comment]
Looks awesome - mergin' :ship:

**Pull request merged**